### ISSUE TITLE
setting that uses `.gitignore` to exclude directories instead of `build.directoryFilters`

### ISSUE THREAD

**Issue by vogtb** [golang/vscode-go#2727]
**tl;dr** -- custom build writes thousands of files to target directories that are ignored by `.gitignore`. But it's not clear that gopls is **not** by default ignoring these, causing it to include _all of my non-source files_, use 100% CPU, and hang indefinitely. The [build.directoryFilters](https://github.com/golang/tools/blob/master/gopls/doc/settings.md#directoryfilters-string) setting is not immediately obvious, and I'd like an option to use `.gitignore` instead, possibly by default.

## Problem

I work with many git submodules in a single large git module that is also my vscode workspace. Building these modules results in thousands files across their own target directories, which are all ignored with `.gitignore` at the workspace level, and individual `.gitignore` files at each git submodule.

When I build these modules _all of them_ are being sent to gopls to be scanned, causing _gopls _use 100% CPU and become unresponsive._

## Solution

A setting that lets me use `.gitignore` to exclude files from being used by gopls. Additionally, I'd like this setting to be on by default, as it's safer and causes fewer surprises to users. This would be an alternative the existing [build.directoryFilters](https://github.com/golang/tools/blob/master/gopls/doc/settings.md#directoryfilters-string) setting.

## Current Alternative

I'm using `build.directoryFilters` for now, but it seems a bit weird that gopls is using "[by default] all directories".

## See also

- Possibly causing #2687.
- Maybe related to #2623.

**Comment by vogtb** [comment]
Running `go build ./...` performs normally.

Using something like vscode's `files.exclude` would also be fine. I thought using `.gitignore` could be a sane default as, in my mind, there's a large overlap between what you don't check into git, and what you don't want scanned. But I also understand the desire to keep things clean, and not have multiple settings affecting what's scanned, or not scanned.

**Comment by hyangah** [comment]
Thanks for testing `go build ./...`. It's interesting that `gopls` falls over while `go build ./...` works ok. Something strange is  going on (cc @findleyr )

One of the complications in translating `files.exclude` or `.gitignore` or other ignore files to `build.directoryFilters` is the pattern syntax difference.
